pkg/controller/replicaset/replica_set.go

```diff
@@ -707,22 +709,30 @@ func (rsc *ReplicaSetController) syncReplicaSet(ctx context.Context, key string)
 	if err != nil {
 		return err
 	}
-	// Ignore inactive pods.
-	filteredPods := controller.FilterActivePods(logger, allPods)
 
-	// NOTE: filteredPods are pointing to objects from cache - if you need to
+	// NOTE: activePods and terminatingPods are pointing to objects from cache - if you need to
 	// modify them, you need to copy it first.
-	filteredPods, err = rsc.claimPods(ctx, rs, selector, filteredPods)
+	var terminatingPods []*v1.Pod
+	activePods := controller.FilterActivePods(logger, allPods)
+	activePods, err = rsc.claimPods(ctx, rs, selector, activePods)
 	if err != nil {
 		return err
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentPodReplacementPolicy) {
+		terminatingPods = controller.FilterTerminatingPods(allPods)
+		terminatingPods, err = rsc.claimPods(ctx, rs, selector, terminatingPods)
```

Comment:
why are we claiming terminating pods? I expected us to only count pods that already belonged to this replicaset

Reply:
I wasn't expecting to change claiming behavior around terminating pods, just counting terminating pods already owned by the replicaset

Reply:
just counting terminating pods already managed by the replicaset is a much smaller change to reason about than starting to claim terminating pods as well... (it's not adding any new API calls, it's just tweaking the data we're already reporting upward into ReplicaSet and Deployment)

Reply:
I do not have a clear use case for this, so I have removed the claiming of terminating pods. Also, the `PodControllerRefManager` does not fully support this use case, as we can only release and not adopt terminating pods.